replica/memtable.cc

```diff
@@ -31,30 +31,26 @@ static mutation_reader make_partition_snapshot_flat_reader_from_snp_schema(
         std::any pointer_to_container,
         streamed_mutation::forwarding fwd, memtable& memtable);
 
-void memtable::memtable_encoding_stats_collector::update_timestamp(api::timestamp_type ts) noexcept {
-    if (ts != api::missing_timestamp) {
-        encoding_stats_collector::update_timestamp(ts);
-        min_max_timestamp.update(ts);
-    }
-}
-
 memtable::memtable_encoding_stats_collector::memtable_encoding_stats_collector() noexcept
     : min_max_timestamp(0, 0)
+    , min_live_timestamp(api::max_timestamp)
+    , min_live_row_marker_timestamp(api::max_timestamp)
 {}
 
 void memtable::memtable_encoding_stats_collector::update(atomic_cell_view cell) noexcept {
-    update_timestamp(cell.timestamp());
+    auto is_live = cell.is_live();
```

Comment:
Without knowing the tombstones above the hierarchy, we don't know for sure whether this cell is live or not. That said, in the worst case we have false-positives here and we will consider more timstamps than we should have. It is safe from correctness POV.